### DISCUSSION TITLE
lang.python nvim-dap-python debugpy missing?

### DISCUSSION THREAD

**Discussion by mstinsky** [LazyVim/LazyVim#1898] [Q&A] [answered]
Hello,

I am trying to get nvim-dap-python working, for that I added lang.python and dap.core as lazyvim extra plugin.
But I am not able to use the debugger as Mason is missing the debugpy package.

Is debugpy suppose to get automatically installed? 
Trying to manually install debugpy through Mason just gives me "Cannot make changes, 'modifiable' is off", so I suppose LazyVim does not want one to install packages manually?

How is this all meant to be configured, does anyone could share his config or help me in the right direction? :)
Thanks in advance.

**Comment by dpetka2001** [comment]
Just press `<leader>db` to set a breakpoint and it should automatically install the debugger.

**Reply by mstinsky** [reply]
Mh sadly it does not install debugpy automatically when I set a breakpoint.

**Reply by dpetka2001** [reply]
Strange, this works for me in a fresh Lazyvim starter template, where I only enable `lang.python` and `dap.core` from LazyExtras. How are you enabling the extras? From the `LazyExtras` UI or in `lua/config/lazy.lua`?

**Reply by mstinsky** [reply]
So, I also tested it with a clean lazyvim starter template and indeed there it is working.
Then I trimmed down everything from my config did a clean install again and when I add the following to my `lua/config/lazy.lua` spec its not working anymore.
```
    { import = "lazyvim.plugins.extras.lang.python" },
    { import = "lazyvim.plugins.extras.formatting.black" },
    { import = "lazyvim.plugins.extras.lang.go" },
    { import = "lazyvim.plugins.extras.lang.terraform" },
    { import = "lazyvim.plugins.extras.lang.docker" },
    { import = "lazyvim.plugins.extras.lang.json" },
    { import = "lazyvim.plugins.extras.lang.markdown" },
    { import = "lazyvim.plugins.extras.coding.yanky" },
    { import = "lazyvim.plugins.extras.dap.core" },
```

**Reply by mstinsky** [reply]
Even more trimmed down:
```
    { import = "lazyvim.plugins.extras.lang.python" },
    { import = "lazyvim.plugins.extras.formatting.black" },
    { import = "lazyvim.plugins.extras.dap.core" },
```
Also does not work to install debugpy.

**Reply by mstinsky** [reply, marked as answer]
aha!
It seems like the order of import is important here.
`{ import = "lazyvim.plugins.extras.dap.core" },` It seems that the dap core has to be imported first for some reason.

**Reply by folke** [reply]
that's correct. Order is important. Better to use `:LazyExtras` so that the order will be correct

**Reply by mstinsky** [reply]
Thanks for confirming @folke
Would you recommend setting everything through `:LazyExtras` and then commiting the `lazyvim.json` to a dotfile repo for nvim to have a defined state when starting from a clean nvim installation?
